### PR TITLE
docs: add production deployment guide

### PR DESCRIPTION
**Pull request by dunglas**
Embed and replace https://github.com/dunglas/frankenphp/pull/261.
Closes https://github.com/dunglas/frankenphp/issues/220.

Text adapted from #261 and https://github.com/dunglas/symfony-docker/blob/main/docs/production.md.

### CODE REVIEW

File: docs/production.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+```dockerfile
+FROM dunglas/frankenphp
+
+# Be sure to replace "your-domain-name.example.com" by your domain name
+ENV SERVER_NAME=your-domain-name.example.com
+# If you want to disable HTTPS, use this value instead:
+#ENV SERVER_NAME=:80
+
+# Enable PHP production settings
+RUN mv "$PHP_INI_DIR/php.ini-production" "$PHP_INI_DIR/php.ini"
```
Comment (withinboredom) [inline docs/production.md:22]: Do we still need to do the `sed` here to change variable order or anything?

File: docs/production.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+Go to `https://your-domain-name.example.com` and enjoy!
+
+> [!CAUTION]
+> Docker can have a cache layer, make sure you have the right build for each deployment or rebuild your project with --no-cache option to avoid cache issue 
+
+## Deploying on Multiple Nodes
+
+If you want to deploy your app on a cluster of machines, you can use [Docker Swarm](https://docs.docker.com/engine/swarm/stack-deploy/),
+which is compatible with the provided Compose files.
+To deploy on Kubernetes, take a look at [the Helm chart provided with API Platform](https://api-platform.com/docs/deployment/kubernetes/), which can be easily adapted for use with Symfony Docker.
```
Comment (withinboredom) [inline docs/production.md:138]: I'd also recommend https://caddyserver.com/docs/modules/caddy.storage.s3#github.com/ss098/certmagic-s3 to store certificates in a shared way across multiple servers. That might be out of scope here, but it works exceptionally well.

Comment (dunglas) [reply docs/production.md:138]: We should do that in the API Platform chart.

**Comment by staudenmeir** [comment]
Hi @dunglas,
The two images aren't loading on the website due to a trailing slash in the image URLs: https://frankenphp.dev/docs/production/

Bildschirmfoto 2024-01-12 um 17 33 49
Bildschirmfoto 2024-01-12 um 17 36 00

The Markdown looks correct, so probably an issue with the HTML generation?

**Pull request merged**